traits/trait_types.py

```diff
@@ -1953,19 +1957,44 @@ def __init__(self, *args, **metadata):
                     "when using the 'values' keyword"
                 )
         else:
-            default_value = args[0]
-            if (len(args) == 1) and isinstance(default_value, EnumTypes):
-                args = default_value
-                default_value = enum_default(args)
-            elif (len(args) == 2) and isinstance(args[1], EnumTypes):
-                args = args[1]
+            if len(args) < 1:
+                raise TraitError("Enum trait requires at "
+                                 "least 1 argument.")
+
+            if len(args) == 1:
+                arg = args[0]
+                if isinstance(arg, EnumTypes):
+                    default_value = enum_default(arg)
+                    self.values = tuple(arg)
+
+                elif isinstance(arg, (str, bytes, bytearray)):
```

Comment:
This branch should have a comment explaining why we're special-casing `str`, `bytes` and `bytearray`.